File: frontend/src/types/recordatorio.ts
export interface Recordatorio {
  id: string;
  procesoId: string;
  titulo: string;
  descripcion: string | null;
  fechaVencimiento: string;
  completado: boolean;
  fechaCreacion: string;
  correoNotificacion: string | null;
  demandanteDelProceso?: string;
  radicadoDelProceso?: string;
}

export interface RecordatorioCreate {
  titulo: string;
  descripcion?: string;
  fechaVencimiento: string;
  correoNotificacion?: string;
}

File: backend/ProcesosApi/Models/Recordatorio.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace ProcesosApi.Models;

public class Recordatorio
{
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = ObjectId.GenerateNewId().ToString();
    public string Titulo { get; set; } = string.Empty;
    public string? Descripcion { get; set; }
    public DateTime FechaVencimiento { get; set; }
    public bool Completado { get; set; }
    public DateTime FechaCreacion { get; set; }
    public string? CorreoNotificacion { get; set; }
    public bool NotificacionEnviada { get; set; }
}
